waiter/src/waiter/util/http_utils.clj

```diff
@@ -40,7 +40,7 @@
 
 (defn spnego-authentication
   "Returns an Authentication$Result for endpoint which will use SPNEGO to generate an Authorization header"
-  [^URI endpoint]
+  [^URI endpoint & {:keys [proxy-auth] :or {proxy-auth false}}]
```

Comment:
Nitpick: we can promote `proxy-auth?` (with question mark) to a required parameter.